### PR TITLE
Add `assign` indirection for `points` to `PointCloud`

### PR DESCRIPTION
**Pull request by kunaltyagi**

### CODE REVIEW

File: common/include/pcl/point_cloud.h [hunk 1]
```diff
@@ -483,6 +483,48 @@ namespace pcl
...
+       * \details The behavior is undefined if either argument is an iterator into
+       * `*this`
+       * \note This breaks the organized structure of the cloud by setting the height to
+       * 1!
+       */
+      template <class InputIt>
+      void
+      assign(InputIt first, InputIt last)
+      {
+        points.assign(first, last);
```
Comment (SergioRAgostinho) [inline common/include/pcl/point_cloud.h:510]: No move here? Are you predicting references being passed?

**Comment by SergioRAgostinho** [comment]
merge after replying.

File: common/include/pcl/point_cloud.h (hunk 1 above)
Comment (kunaltyagi) [reply common/include/pcl/point_cloud.h:510]: Iterators are supposed to be lightweight, but I see your point

**Comment by SergioRAgostinho** [comment]
Squash and merge after CI runs gree.

**Pull request merged**